model/notification.go

```diff
@@ -1,12 +1,95 @@
 package model
 
+import (
+	"fmt"
+	"log"
+
+	"github.com/jmoiron/sqlx"
+)
+
 type Notification struct {
+	// From the notifications table.
 	UserID         int    `db:"user_id"`
 	AnnouncementID int    `db:"announcement_id"`
 	Status         string `db:"status"`
 	Receipt        string `db:"receipt"`
-	ReceiptStatus  string `db:"receipt_status"`
-	Timestamp      string `db:"timestamp"`
+	// From joined tables.
+	ExpoPushToken string `db:"expo_push_token"`
+	Title         string `db:"title"`
+	Body          string `db:"body"`
 }
 
-// TODO
+func EnqueueAnnouncementNotifications(db *sqlx.DB) {
+	log.Println("Starting to enqueue announcement notifications.")
+
+	tx := db.MustBegin()
```

Comment:
Does sqlx have a function for running something as a transaction? If not, I'd write something like:

```
func transact(db *sqlx.DB, do func(tx *sqlx.Tx) error) error {
    tx := db.MustBegin()
    if err := do(tx); err != nil {
      tx.Rollback()
      return err
    }
    return tx.Commit()
}
```

then you can rewrite the code below more like:

```
transact(db, func(tx *sqlx.Tx) error {
    if err := ...; err != nil {
        return err
    }
    ...
    return nil
})
```

Reply:
I don't believe it does, but good thinking.

Reply:
This is done in the next commit.